ql/src/java/org/apache/hadoop/hive/ql/io/RecordReaderWrapper.java

```diff
@@ -69,7 +70,14 @@ static RecordReader create(InputFormat inputFormat, HiveInputFormat.HiveInputSpl
       JobConf jobConf, Reporter reporter) throws IOException {
     int headerCount = Utilities.getHeaderCount(tableDesc);
     int footerCount = Utilities.getFooterCount(tableDesc, jobConf);
-    RecordReader innerReader = inputFormat.getRecordReader(split.getInputSplit(), jobConf, reporter);
+
+    RecordReader innerReader = null;
+    try {
+     innerReader = inputFormat.getRecordReader(split.getInputSplit(), jobConf, reporter);
+    } catch (InterruptedIOException iioe) {
+      // If reading from the underlying record reader is interrupted, return a no-op record reader
+      return new ZeroRowsInputFormat().getRecordReader(split.getInputSplit(), jobConf, reporter);
```

Comment:
why is it better to return with no-op record reader instead of letting this codepath fail and handle the exception somewhere else? doesn't this mask issues?

Reply:
Hey.

So, in my experimentation, this is the least-bad option.  I did this to preserve the previous behavior.  The Hive code is not setup to handle this error condition.  As thing currently stand in `master`, if the calling Thread was interrupted, the thread would finish fetching the rows regardless and then just later ignore them (throw them away).  The calling code does not handle 'null' return value and it does not handle this Exception.  As currently implemented in Hive `master`, if it gets an exception it simply exits execution with an Error message, without implementing a lot more code, there is no way to ignore/skip this one specific error type.  So, the cleanest thing to do is to return `ZeroRows` since it's going to be thrown away later anyway.